todo-app/src/App.js

```diff
@@ -1,13 +1,83 @@
-import React, { Component } from 'react';
+//jshint esnext: true
+
+import React from 'react';
+
+import TodoList from './Components/todoList';
+import AddTodoItem from './Components/AddTodoItem';
+import './App.css';
+
+
+export default class App extends React.Component {
+
+  
+  state= {
+    todos: [],
+    isEditing: null
+  }
+
+
+  handleToggleEditing = (id) => {
+    this.setState({ isEditing: id || null })
```

Comment:
This is an interesting choice. It is a pattern that used to be pretty common (maybe it still is) - if you do not include an argument, the function becomes a "cancel" action.

This might work in some cases, but I think it's a bad idea here. Because this function is passed down through several children, it should be immediately clear what it does to limit confusion. I'd recommend either adding a second argument, like:

    handleToggleEditing = (id, cancelEditing: boolean) =>

Or just declare a second cancel function in a single line, like:

    handleCancelEditing = () => this.handleToggleEditing(null)